src/hypothesis/strategies.py

```diff
@@ -461,9 +462,35 @@ def sampled_from(elements):
     1 values with 10, and sampled_from((1, 10)) will shrink by trying to
     replace 10 values with 1.
     """
-    from hypothesis.searchstrategy.misc import SampledFromStrategy
-    from hypothesis.internal.conjecture.utils import check_sample
-    values = check_sample(elements)
+    try:
```

Comment:
I’d put this in `hypothesis.internal.validation`, which is where a lot of other strategy validation logic lives (e.g. `check_valid_bound()` in `integers()`).

The original `check_samples()` was arguably in the wrong place too, but that’s less important now.